Gems/SurfaceData/Code/Source/TerrainSurfaceDataSystemComponent.cpp

```diff
@@ -168,12 +152,13 @@ namespace SurfaceData
                     const float terrainHeight = terrain->GetHeight(inPosition, AzFramework::Terrain::TerrainDataRequests::Sampler::BILINEAR, &isTerrainValidAtPoint);
```

Comment:
Could use AZStd::optional<float> for this so you don't have to bother with a flag.

Reply:
Hmmm, that's an interesting idea, but also might make the API harder to use because you wouldn't be able to just use the result in the middle of math.  We should definitely discuss as a team and see if we want to do this as a separate PR.

Reply:
You could still just use the result in math, you'd just need to deref it like a pointer. Pretty sure it's 0 overhead. Anyway just an option, no need to do it now.